src/tools/pkcs11-tool.c

```diff
@@ -3378,7 +3393,25 @@ gen_key(CK_SLOT_ID slot, CK_SESSION_HANDLE session, CK_OBJECT_HANDLE *hSecretKey
 			FILL_ATTR(keyTemplate[n_attr], CKA_KEY_TYPE, &key_type, sizeof(key_type));
 			n_attr++;
 		}
-		else {
+
+		else if (strncasecmp(type, "HKDF:", strlen("HKDF:")) == 0) {
+			CK_MECHANISM_TYPE mtypes[] = {CKM_HKDF_KEY_GEN};
+			size_t mtypes_num = sizeof(mtypes) / sizeof(mtypes[0]);
```

Comment:
You could use ARRAY_SIZE macro for this

Reply:
The code I added for HKDF is a copy/paste of the GENERIC case.
We would have to do the same for the other key types otherwise the code would not be consistent.
I will address your remark in a new PR.